src/recantingwitness.jl

```diff
@@ -0,0 +1,51 @@
+export has_recanting_witness
+function has_recanting_witness(g::AbstractGraph{T}, u::Integer, v::Integer,  blocked_edges::AbstractGraph{T}) where T
+    on_blocked = zeros(Bool, nv(g))
+    on_open = zeros(Bool, nv(g))
+    may_blocked = zeros(Bool, nv(g))
+    for e in edges(blocked_edges) # mark excluded vertices as seen
+        may_blocked[src(e)] = true
+    end
+    u == v && error("u == v")
+
+    next_open = Vector{T}()
+    next_blocked = Vector{T}()
+    
+    push!(next_open, u)
+    on_open[u] = true
+    while !isempty(next_open)
+        src = popfirst!(next_open) # get new element from queue
+        for vertex in outneighbors(g, src)
+            if !on_open[vertex]
```

Comment:
Removing this condition works for all the test cases. I think on_blocked for vertex `v` isn't becoming true when an open path is already found.

Reply:
This is about recursing twice down an open path. Think of a vertex which is reachable through two open path. Then we don't want to check twice for all path reachable. Or do I miss something?